notebook/nbconvert/handlers.py

```diff
@@ -76,40 +76,50 @@ def get_exporter(format, **kwargs):
 class NbconvertFileHandler(IPythonHandler):
 
     SUPPORTED_METHODS = ('GET',)
-    
+
     @web.authenticated
     def get(self, format, path):
-        
+
         exporter = get_exporter(format, config=self.config, log=self.log)
-        
+
         path = path.strip('/')
         # If the notebook relates to a real file (default contents manager),
         # give its path to nbconvert.
         if hasattr(self.contents_manager, '_get_os_path'):
             os_path = self.contents_manager._get_os_path(path)
+            ext_resources_dir, basename = os.path.split(os_path)
         else:
-            os_path = ''
+            ext_resources_dir = None
 
         model = self.contents_manager.get(path=path)
         name = model['name']
         if model['type'] != 'notebook':
             # not a notebook, redirect to files
             return FilesRedirectHandler.redirect_to_files(self, path)
 
+        nb = model['content']
+
         self.set_header('Last-Modified', model['last_modified'])
 
+        # create resources dictionary
+        mod_date = model['last_modified'].strftime(text.date_format)
+        nb_title = nb.metadata.get("title","") or os.path.splitext(name)[0]
```

Comment:
Not convinced about the title from notebook metadata taking precedence over the filename used to identify it. The notebook metadata is already available in the template, it doesn't need to be added to resources.

Reply:
Fine, this was a fix that I had added on the nbconvert side for the LaTeX interpreter and was just trying to keep it consistent here. But I'm not wedded to it.